internal/weaver/logger.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package weaver
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/ServiceWeaver/weaver/runtime/logging"
+	"github.com/ServiceWeaver/weaver/runtime/protos"
+)
+
+// remoteLogger sends log entries to a specified component method.
+//
+// A go routine makes a sequence of calls to the logger component (at most one
+// call is outstanding at a time). Log entries that arrive while a call is in
+// progress are buffered together and sent in the next call.
+type remoteLogger struct {
+	c        chan *protos.LogEntry
+	fallback io.Writer              // Fallback destination when dst() returns an error
+	pp       *logging.PrettyPrinter // Used when sending to dst fails
+}
+
+const logBufferCount = 1000
+
+func newRemoteLogger(fallback io.Writer) *remoteLogger {
+	rl := &remoteLogger{
+		c:        make(chan *protos.LogEntry, logBufferCount),
+		fallback: fallback,
+		pp:       logging.NewPrettyPrinter(false),
+	}
+	return rl
+}
+
+func (rl *remoteLogger) send(entry *protos.LogEntry) {
```

Comment:
`send -> log` ? (since we aren't sending yet.)

Reply:
Done.